src/svc_vc.c

```diff
@@ -658,10 +658,13 @@ svc_rdvs_destroy(SVCXPRT *xprt, u_int flags, const char *tag, const int line)
 	mem_free(rdvs, sizeof(struct cf_rendezvous));
 	mem_free(xprt, sizeof(SVCXPRT));
 
-	refcnt = rpc_dplx_unref(rec,
-				RPC_DPLX_FLAG_LOCKED | RPC_DPLX_FLAG_UNLOCK);
-	if (!refcnt)
-		mem_free(xd, sizeof(struct x_vc_data));
```

Comment:
On Di, 2016-09-06 at 10:28 -0700, Daniel Gryniewicz wrote:
> In src/svc_vc.c:
> > @@ -658,10 +658,13 @@ svc_rdvs_destroy(SVCXPRT *xprt, u_int flags,
> const char *tag, const int line)
> >  	mem_free(rdvs, sizeof(struct cf_rendezvous));
> >  	mem_free(xprt, sizeof(SVCXPRT));
> >  
> > -	refcnt = rpc_dplx_unref(rec,
> > -				RPC_DPLX_FLAG_LOCKED |
> RPC_DPLX_FLAG_UNLOCK);
> > -	if (!refcnt)
> > -		mem_free(xd, sizeof(struct x_vc_data));
> This part looks correct.
> —
No, the returned value is the refcount for rec and got nothing to do
with xd.
We need to check for xd's refcount and free it if it's zero.
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   > 
>   > 

> 
> 

> 
> 
> 
<span class="-x-evo-quoted"><span class="-x-evo-quote-character">> </span></span>{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/nfs-ganesha/ntirpc","title":"nfs-ganesha/ntirpc","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/nfs-ganesha/ntirpc"}},"updates":{"snippets":[{"icon":"PERSON","message":"@dang in #18: This part looks correct."}],"action":{"name":"View Pull Request","url":"https://github.com/nfs-ganesha/ntirpc/pull/18/files/b4254b923aaa4847e6732d4ff893614ac2558388#r77680766"}}}